From 812f98536245a501b6d3b389afe01306ea32a84c Mon Sep 17 00:00:00 2001
From: Max Bires <jbires@google.com>
Date: Wed, 17 Jun 2020 06:03:54 -0700
Subject: [PATCH 13/15] Removing authority key identifier OID extension

This is no longer used and needs to be removed from the reference
Keymaster implementation.

Bug: 149746301
Test: atest system/keymaster
Change-Id: I6970b7d2ae54481ba30729d105599d42953efc3f
---
 km_openssl/attestation_utils.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/km_openssl/attestation_utils.cpp b/km_openssl/attestation_utils.cpp
index 6b60fe8..1ed523b 100644
--- a/km_openssl/attestation_utils.cpp
+++ b/km_openssl/attestation_utils.cpp
@@ -416,16 +416,6 @@ keymaster_error_t generate_attestation_common(
                    nullptr,             // crl
                    0 /* flags */);
 
-    X509_EXTENSION_Ptr auth_key_id(X509V3_EXT_nconf_nid(nullptr,           // conf
-                                                        x509v3_ctx.get(),  //
-                                                        NID_authority_key_identifier,
-                                                        const_cast<char*>("keyid:always")));
-    if (!auth_key_id.get() || !X509_add_ext(certificate.get(),  //
-                                            auth_key_id.get(),  // Don't release; copied
-                                            -1 /* insert at end */)) {
-        return TranslateLastOpenSslError();
-    }
-
     if (!X509_sign(certificate.get(), sign_key.get(), EVP_sha256()))
         return TranslateLastOpenSslError();
 
-- 
2.17.1

